At the time of analysis, no genomes of Breviolum psygmophilum were available. A dummy genome was made from an available Transcriptome (Rivera et al.) using a custom perl script <https://github.com/z0on/2bRAD_denovo/blob/master/concatFasta.pl> specifying 97 chromosomes (LaJeunesse et al. 2005). The dummy genome of B. psygmophilum and Astrangia poculata {ref} were then concatenated to form a holobiont reference genome. Raw sequence files were then trimmed to remove Illumina TruSeq adapters and poly-A tails using the fastx-Toolkit (v 0.0.14, Gordon & Hannon, (2010) fastx-Toolkit.[http://hannonlab.cshl.edu/fastx_toolkit.)](http://hannonlab.cshl.edu/fastx_toolkit.)) and resulting sequences that were \<20 bp in length were also removed. Sequences with \>90% of bases having a quality score \>20 were retained,Â  PCR duplicates were removed and resulting quality-filtered reads were mapped to the holobiont genome using bowtie2 (Langmead & Salzberg, 2012). The mapped reads were then assigned into a raw counts file using a custom perl script. There was an average of 634 657 mapped reads with three individuals failing to sequence properly. Of the remaining samples, the mapping efficiencies ranged from 37%-83% with an average mapping efficiency of 70% (Table SX). Reads that were assigned to the photobiont were then discarded and the cleaned coral reads were used for subsequent analyses.